azure/services/aksextensions/aksextensions.go

```diff
@@ -41,10 +45,17 @@ type Service struct {
 // New creates a new service.
 func New(scope AKSExtensionScope) *Service {
 	svc := aso.NewService[*asokubernetesconfigurationv1.Extension, AKSExtensionScope](serviceName, scope)
+	svc.ListFunc = list
 	svc.Specs = scope.AKSExtensionSpecs()
 	svc.ConditionType = infrav1.AKSExtensionsReadyCondition
 	return &Service{
 		Scope:   scope,
 		Service: svc,
 	}
 }
+
+func list(ctx context.Context, client client.Client, opts ...client.ListOption) ([]*asokubernetesconfigurationv1.Extension, error) {
```

Comment:
nit: Can you name this to something more descriptive like `listAKSExtensions`?

Reply:
Does the signature and this existing in the `aksextensions` package make that clear? I've been trying to optimize these for copy-paste-ability since it's pretty much the same between all the ASO services so adding the type again here would be one more place to change.

Reply:
sounds good to me!